crates/typst/src/eval/binding.rs

```diff
@@ -89,23 +89,30 @@ fn destructure_array<F>(
 where
     F: Fn(&mut Vm, ast::Expr, Value) -> SourceResult<()>,
 {
+    let destruct_len = destruct.items().count();
```

Comment:
why was this moved up here? it doesn't seem to be used anywhere but `Spread`, but now we unnecessarily traverse the destructuring AST twice.

Reply:
This is indeed unnecessary.